lib/webapi/webapi.go

```diff
@@ -1298,33 +1295,36 @@ func (m *Handler) shrinkSite(w http.ResponseWriter, r *http.Request, p httproute
 	return siteShrinkOutput{Operation: *operation}, nil
 }
 
-// getSite retrieves details on the specified site
-//
-// GET /portalapi/v1/sites/:domain
+// getCluster returns the specified cluster object.
 //
-// Input: site_id
+//   GET /portalapi/v1/sites/:domain?shallow=(true|false)
 //
 // Output:
-// {
-//   "id": "344238abcd7"
-//   "created": "2016-05-14 13:00:05"
-//   "domain_name": "example.com"
-//   "account_id": "1ab238a8cd5"
-//   "state": "active"
-//   "provisioner": "aws_terraform"
-//   "app": {"package": "gravitational.io/test:1.0.0", "manifest": <...application manifest...>}
-// }
-func (m *Handler) getSite(w http.ResponseWriter, r *http.Request, p httprouter.Params, context *AuthContext) (interface{}, error) {
-	siteDomain := p[0].Value
-	site, err := context.Operator.GetSite(ops.SiteKey{
-		SiteDomain: siteDomain,
-		AccountID:  context.User.GetAccountID(),
-	})
+//
+//   webCluster
+//
+// If 'shallow' flag is true, returns stripped down cluster objects that do
+// not include raw manifest data, icons and other verbose fields.
+func (m *Handler) getCluster(w http.ResponseWriter, r *http.Request, p httprouter.Params, context *AuthContext) (interface{}, error) {
+	shallow, err := getBoolFlag(r, "shallow", false)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	return site, nil
+	key, err := clusterKey(context, p)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	cluster, err := context.Operator.GetSite(*key)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	releases, err := getReleases(context.Operator, *cluster)
+	if err != nil {
+		m.Errorf("Failed to retrieve releases information for cluster %v: %v.",
```

Comment:
Does this need a return?

Reply:
I was intending for this to be a non-critical failure (so cluster UI would still load if releases retrieval fails).